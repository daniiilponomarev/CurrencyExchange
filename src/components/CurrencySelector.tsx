import React, { ChangeEvent } from "react";
// @ts-ignore
import CurrencyTextField from "@unicef/material-ui-currency-textfield";
import FormControl from "@mui/material/FormControl";
import Select, { SelectChangeEvent } from "@mui/material/Select";
import MenuItem from "@mui/material/MenuItem";
import Typography from "@mui/material/Typography";

import { CURRENCIES } from "../constants";
import { formControlStyle } from "../containers/currencyExchange/CurrencyExchangeStyles";
import { CurrencyName } from "../types";

type CurrencySelectorProps = {
  balance: number;
  currency: CurrencyName;
  amount: number;
  maximumInputValue: string;
  handleSelectChange: (event: SelectChangeEvent) => void;
  handleInputChange: (
    event: ChangeEvent<HTMLInputElement>,
    value: number
  ) => void;
};

export const CurrencySelector = ({
  balance,
  currency,
  amount = 0,
  handleSelectChange,
  handleInputChange,
  maximumInputValue,
}: CurrencySelectorProps) => {
  return (
    <>
      <FormControl fullWidth sx={formControlStyle}>
        <Select value={currency} onChange={handleSelectChange}>
          {CURRENCIES.map((currency) => (
            <MenuItem key={currency} value={currency}>
              {currency}
            </MenuItem>
          ))}
        </Select>
        <CurrencyTextField
          label="Amount"
          variant="outlined"
          value={amount}
          currencySymbol=""
          onChange={handleInputChange}
          maximumValue={maximumInputValue}
          minimumValue={"0"}
        />
        <Typography variant="subtitle1">Balance: {balance.toFixed(2)}</Typography>
      </FormControl>
    </>
  );
};
