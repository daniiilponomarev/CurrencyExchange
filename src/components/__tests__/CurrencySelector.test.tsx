import React from "react";
import { render } from "@testing-library/react";
import { CurrencySelector } from "../CurrencySelector";

test("renders learn react link", () => {
  const component = render(
    <CurrencySelector
      balance={1}
      currency="USD"
      amount={2}
      handleSelectChange={() => {}}
      handleInputChange={() => {}}
      maximumInputValue={"5"}
    />
  );

  expect(component).toMatchSnapshot();
});
