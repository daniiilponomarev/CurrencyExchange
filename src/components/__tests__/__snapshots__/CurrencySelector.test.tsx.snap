// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-999iei-MuiFormControl-root"
      >
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            role="button"
            tabindex="0"
          >
            USD
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value="USD"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-nnbavb"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
            data-shrink="true"
          >
            Amount
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
              />
            </div>
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input CurrencyTextField-textField-1 CurrencyTextField-textField-2 MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-5 PrivateNotchedOutline-legendNotched-6"
              >
                <span>
                  Amount
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
        >
          Balance: 
          1.00
        </h6>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth css-999iei-MuiFormControl-root"
    >
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          role="button"
          tabindex="0"
        >
          USD
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="USD"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-nnbavb"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
          data-shrink="true"
        >
          Amount
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart"
        >
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionStart"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            />
          </div>
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input CurrencyTextField-textField-1 CurrencyTextField-textField-2 MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-5 PrivateNotchedOutline-legendNotched-6"
            >
              <span>
                Amount
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
      >
        Balance: 
        1.00
      </h6>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
